cli/cliutil/hostname.go

```diff
@@ -0,0 +1,40 @@
+package cliutil
+
+import (
+	"os"
+	"strings"
+	"sync"
+
+	"golang.org/x/xerrors"
+)
+
+var (
+	hostname     string
+	hostnameOnce sync.Once
+)
+
+// Hostname returns the hostname of the machine, lowercased,
+// with any trailing domain suffix stripped.
+// It is cached after the first call.
+// If the hostname cannot be determined, this will panic.
+func Hostname() string {
+	hostnameOnce.Do(func() { hostname = getHostname() })
+	return hostname
+}
+
+func getHostname() string {
+	h, err := os.Hostname()
+	if err != nil {
+		// Something must be very wrong if this fails.
+		panic(xerrors.Errorf("lookup hostname: %w", err))
```

Comment:
How about choosing a default like `unknown` instead of panic'ing?

Reply:
good ol' localhost perhaps? :D